listenbrainz/webserver/static/js/src/utils.tsx

```diff
@@ -102,4 +103,44 @@ const getPlayButton = (listen: any, onClickFunction: () => void) => {
   );
 };
 
-export { searchForSpotifyTrack, getArtistLink, getTrackLink, getPlayButton };
+const preciseTimestamp = (listened_at: number): string => {
+  const listenDate: Date = new Date(listened_at);
```

Comment:
With the comment above came the realization that we should probably add one last protection in case the expected  ISO date string or unix timestamp is not what we expect, or if an unexpected `undefined` is passed:

We can easily check if the Date object was able to create a new valid date like this:
```
if(isNaN(listenDate)){
    return '—';
}
```

Here I'm suggesting we return a placeholder rather than nothing if the date is invalid. How does that sound? Or should we `return listened_at;` instead?

Reply:
I went with returning listened_at: all numbers will translate into valid outputs (even negative numbers for some reason) so I imagine that if the date is invalid it would be from an incorrectly formatted ISO string that might be more helpful than just a placeholder.